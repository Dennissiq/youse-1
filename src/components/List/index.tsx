import React from 'react'
import { CSSTransition } from 'react-transition-group'
import PersonDiv from '../Person'
import Person from '../../interfaces/Person'
import List from './styles'

export default ({
  filtredItems,
  isLoading
}: {
  filtredItems: Array<Person>
  isLoading: boolean
}) => (
  <List>
    <PersonDiv
      modifier={0}
      header
      item={{ id: 0, title: 'Name', description: 'Email' }}
    />
    {filtredItems.map((item: Person, key: number) => (
      <CSSTransition
        in={!isLoading}
        timeout={{
          enter: key * 100,
          exit: 500
        }}
        key={item.id}
      >
        <PersonDiv header={false} modifier={key} item={item} />
      </CSSTransition>
    ))}
  </List>
)
