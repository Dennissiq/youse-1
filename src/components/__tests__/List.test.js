import React from 'react'
import renderer from 'react-test-renderer'
import List from '../List'

it('renders correctly', () => {
  const filtredItems = [
    {
      description: 'Ahmad31@yahoo.com',
      id: 0,
      title: 'Estefania Kunze',
    },
  ]

  const tree = renderer
    .create(<List isLoading={false} filtredItems={filtredItems} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
