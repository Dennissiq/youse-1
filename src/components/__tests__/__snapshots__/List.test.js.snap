// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="sc-bdnylx jQvQpK header"
  >
    <article
      className="sc-gtssRu dLuDwB"
    >
      <h3
        className="sc-dlnjPT eRMVCU"
      >
        Name
      </h3>
      <span
        className="sc-hKFyIo hiuoVs"
      >
        Email
      </span>
    </article>
  </div>,
  <div
    className="sc-bdnylx kIJtuq"
  >
    <article
      className="sc-gtssRu dLuDwB"
    >
      <h3
        className="sc-dlnjPT ibOGxv"
      >
        Estefania Kunze
      </h3>
      <span
        className="sc-hKFyIo beMXmz"
      >
        Ahmad31@yahoo.com
      </span>
    </article>
  </div>,
]
`;
